python/ray/dashboard/agent.py

```diff
@@ -163,6 +163,10 @@ def http_session(self):
     def get_node_id(self) -> str:
         return self.node_id
 
+    @property
+    def is_head(self) -> bool:
+        return self.ip == parse_address(self.gcs_address)[0]
```

Comment:
I think this is not reliable if people are using the ray `cluster_utils` utility to run multiple logical nodes on the same host

The most reliable identity check would be using the node ID